tests/App/config/services.yml

```diff
@@ -1,10 +1,20 @@
 services:
+    _defaults:
+        autowire: true
+        autoconfigure: true
+
     # Under real conditions, this service is registered by "doctrine/doctrine-bundle"
     database_connection:
         class: Sonata\AdminBundle\Tests\Fixtures\DoctrineBundle\Connection
 
     Sonata\AdminBundle\Tests\App\Model\FooRepository: ~
 
+    Sonata\AdminBundle\Controller\CRUDController:
```

Comment:
This should not be needed

Reply:
I don't know how to avoid this line. Do you have an idea ?

Reply:
What error do you get if you dont add this lines?

Reply:
Uncaught PHP Exception InvalidArgumentException: "The controller for URI "/admin/tests/app/foo/list" is not callable: Controller "Sonata\AdminBundle\Controller\CRUDController" has required constructor arguments and does not exist in the container. Did you forget to define the controller as a service?"

Reply:
Can you look at that route definition? it should be in a test file, you need to point the controller part to the id of the controller you have just defined

Reply:
I don't think is possible since we add kernelDebug instance of Kernel object in the constructor

Reply:
But you added kernel debug in:

```
<service id="sonata.admin.controller.crud" class="Sonata\AdminBundle\Controller\CRUDController" public="true">
```

Reply:
i think Test Kernel doesn't load core.xml ?

Reply:
It should be loaded since we enable the SonataAdminBundle on the testCase. You need to modify this:

https://github.com/sonata-project/SonataAdminBundle/blob/master/src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L70

and use the id:  `sonata.admin.controller.crud`, can you try?

Reply:
New error message : "sonata.admin.controller.crud" has no container set, did you forget to define it as a service subscriber?

I try other things but i'm not confident

Reply:
I fixed thanks to your information and this line : https://github.com/sonata-project/SonataAdminBundle/pull/6093/files#diff-ea95afcce69fc04cf8246bfd1d13ae75R4